connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/aiagent/agent/DefaultAiAgentRequestHandler.java

```diff
@@ -0,0 +1,158 @@
+import io.camunda.connector.agenticai.aiagent.tools.ToolCallingHandler;
+import io.camunda.connector.api.error.ConnectorException;
+import io.camunda.connector.api.outbound.OutboundConnectorContext;
+import java.util.Optional;
+
+public class DefaultAiAgentRequestHandler implements AiAgentRequestHandler {
+
+  private static final int DEFAULT_MAX_MODEL_CALLS = 10;
+  private static final int DEFAULT_MAX_MEMORY_MESSAGES = 20;
+
+  private final ObjectMapper objectMapper;
+  private final ChatModelFactory chatModelFactory;
+  private final ToolCallingHandler toolCallingHandler;
+
+  public DefaultAiAgentRequestHandler(
+      ObjectMapper objectMapper,
+      ChatModelFactory chatModelFactory,
+      ToolCallingHandler toolCallingHandler) {
+    this.objectMapper = objectMapper;
+    this.chatModelFactory = chatModelFactory;
+    this.toolCallingHandler = toolCallingHandler;
+  }
+
+  @Override
+  public AgentResponse handleRequest(OutboundConnectorContext context, AgentRequest request) {
+    final AgentRequest.AgentRequestData requestData = request.data();
+    final AgentContext agentContext =
+        Optional.ofNullable(requestData.context()).orElseGet(AgentContext::empty);
+
+    // initialize configured model
+    final ChatLanguageModel chatModel = chatModelFactory.createChatModel(request);
+
+    // set up memory and load from context if available
+    final AgentContextChatMemoryStore chatMemoryStore =
+        new AgentContextChatMemoryStore(objectMapper);
+    chatMemoryStore.loadFromAgentContext(agentContext);
+
+    final ChatMemory chatMemory =
+        MessageWindowChatMemory.builder()
+            .maxMessages(
+                Optional.ofNullable(requestData.memory().maxMessages())
+                    .orElse(DEFAULT_MAX_MEMORY_MESSAGES))
+            .chatMemoryStore(chatMemoryStore)
+            .build();
+
+    // check configured guardrails
+    checkGuardrails(requestData, agentContext);
+
+    // update memory with system + new user messages/tool call responses
+    addSystemPromptIfNecessary(chatMemory, requestData);
+    addUserMessagesFromRequest(agentContext, chatMemory, requestData);
+
+    // call LLM API with updates messages + resolved tool specifications
+    final var toolSpecifications =
+        toolCallingHandler.loadToolSpecifications(
+            context.getJobContext().getProcessDefinitionKey(),
+            requestData.tools().containerElementId());
+
+    final var chatRequest =
+        ChatRequest.builder()
+            .messages(chatMemory.messages())
+            .toolSpecifications(toolSpecifications)
+            .build();
+
+    final ChatResponse chatResponse = chatModel.chat(chatRequest);
+    final AiMessage aiMessage = chatResponse.aiMessage();
+    chatMemory.add(aiMessage);
+
+    // extract tool call requests from LLM response
+    final var toolsToCall = toolCallingHandler.extractToolsToCall(toolSpecifications, aiMessage);
+    final var nextAgentState =
```

Comment:
Very simple "state machine" to determine if we need to inject a user/follow-up prompt or a tool calling result in the next iteration (see `addUserMessagesFromRequest`).